File: array/decompress-run-length-encoded-list/decompress-run-length-encoded-list.py
class Solution:
    def decompressRLElist(self, nums):
        arr = []
        for i in range(0, len(nums) - 1, 2):
            while nums[i]:
                arr.append(nums[i + 1])
                nums[i] = nums[i] - 1

        return arr

        

File: array/decompress-run-length-encoded-list/decompress-run-length-encoded-list.ts
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var decompressRLElist = function (nums: Array<number>) {
    let arr = [];
    for (let i = 0; i < nums.length; i = i + 2) {
        arr.push(...(new Array(nums[i])).fill(nums[i + 1]));
    }

    return arr;
};
